pkg/raft/transport/helpers.go

```diff
@@ -0,0 +1,90 @@
+package transport
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/hashicorp/raft"
+)
+
```

Comment:
can we have a big comment here explaining this file and the types? lik what do appendentries/requestvote etc. do and what cases are they used for